header.php

```diff
@@ -44,8 +44,39 @@
 			<?php endif; ?>
 		</div><!-- .site-branding -->
 
-		<nav id="site-navigation" class="main-navigation">
-			<button class="menu-toggle" aria-controls="primary-menu" aria-expanded="false"><?php esc_html_e( 'Primary Menu', '_s' ); ?></button>
+		<nav id="site-navigation" class="main-navigation"
+			<?php if ( _s_is_amp() ) : ?>
+				[class]=" siteNavigationMenu.expanded ? 'main-navigation toggled' : 'main-navigation' "
+			<?php endif; ?>
+		>
+			<?php if ( _s_is_amp() ) : ?>
+				<amp-state id="siteNavigationMenu">
+					<script type="application/json">
+						{
+							"expanded": false
+						}
+					</script>
+				</amp-state>
+				<input
+					id="site-navigation-expanded"
+					type="checkbox"
+					hidden
+					on="change:AMP.setState( { siteNavigationMenu: { expanded: event.checked } } )"
+				>
```

Comment:
This hidden checkbox is far from ideal. It's mainly here because AMP doesn't yet support a `toggleClass` action. See https://github.com/ampproject/amphtml/issues/1032